src/app/components/Dropdown.tsx

```diff
@@ -52,20 +52,30 @@ const visuallyHidden = css`
   clip: rect(1px, 1px, 1px, 1px);
 `;
 
-type Props = React.PropsWithChildren<{
+interface ControlledProps {
+  open: boolean;
+  setOpen: (value: boolean) => void;
+}
+
+interface Props {
   toggle: React.ReactNode;
   className?: string;
   onToggle?: () => void;
-}>;
+}
 
 // tslint:disable-next-line:variable-name
-const TabHiddenDropDown = styled(({toggle, children, className, onToggle}: Props) => {
-  const [open, setOpen] = React.useState<boolean>(false);
+const TabHiddenDropDown = styled((
+  {toggle, children, className, onToggle, ...props}: React.PropsWithChildren<Props | Props & ControlledProps>
+) => {
+  const { open: controlledOpen, setOpen: controlledSetOpen } = props as Props & ControlledProps;
+  const [open, setOpenState] = React.useState<boolean>(false);
+  const isOpen = controlledOpen !== undefined ? controlledOpen : open;
+  const setOpen = controlledSetOpen !== undefined ? controlledSetOpen : setOpenState;
```

Comment:
Seems complicated for a logic of opening and closing the `Dropdown`.
Is `const [open, setOpenState] = React.useState<boolean>(false);` really necessary?
And the naming creates a confusion. One booleans have `is` prefix, but other don't.

Reply:
Previously the state of this component was handled only internally, but know we need to be able to control the open/close state for this component from outside of the component and keep the internal solution for previous situations - this makes this component a controlled and uncontrolled at the same time

For simplicity of the whole component I've decided to use everywhere  `isOpen` and `setOpen` variables and lines 70 - 74 are making sure that these variables point to the proper state and set state handlers